Doc/library/_thread.rst

```diff
@@ -61,15 +61,28 @@ This module defines the following constants and functions:
       :func:`sys.unraisablehook` is now used to handle unhandled exceptions.
 
 
-.. function:: interrupt_main()
+.. function:: interrupt_main([signum])
 
-   Simulate the effect of a :data:`signal.SIGINT` signal arriving in the main
-   thread. A thread can use this function to interrupt the main thread.
+   Simulate the effect of a signal arriving in the main thread.
+   A thread can use this function to interrupt the main thread.
 
-   If :data:`signal.SIGINT` isn't handled by Python (it was set to
+   If given, *signum* is the number of the signal to simulate.
+   If *signum* is not given, :data:`signal.SIGINT` is simulated.
+
+   If the given signal isn't handled by Python (it was set to
    :data:`signal.SIG_DFL` or :data:`signal.SIG_IGN`), this function does
    nothing.
 
+   .. versionchanged:: 3.10
+      The *signum* argument is added to customize the signal number.
+
+   .. note::
+      This does not emit the corresponding signal, so you can simulate
+      signals which would otherwise stop or kill the process, such as
+      :data:`signal.SIGSTOP`.
```

Comment:
```suggestion
      This does not emit the corresponding signal but calls the associated handler
      (if it exists) so you can simulate signals which would otherwise stop or kill the
      process, such as :data:`signal.SIGSTOP`.
```

Reply:
Won't you get a `OSError` if you register a handler for a signal that skips user space such as `SIGSTOP`?

Reply:
I don't know, to be honest.

Reply:
You're right, `OSError` seems to be raised.

Reply:
I tweaked the wording but removed the part that mentioned simulating uncatchable signals.